source/mir/math/sum.d

```diff
@@ -190,48 +190,57 @@ nothrow @nogc unittest
 
 /// Moving mean
 version(mir_test)
+@safe pure nothrow @nogc
 unittest
 {
-    import mir.ndslice.topology: linspace;
+    import mir.internal.utility: isFloatingPoint;
     import mir.math.sum;
-    import mir.array.allocation: array;
+    import mir.ndslice.topology: linspace;
+    import mir.rc.array: rcarray;
 
-    class MovingAverage
+    struct MovingAverage(T)
+        if (isFloatingPoint!T)
     {
-        Summator!(double, Summation.precise) summator;
+        import mir.math.stat: MeanAccumulator;
+        import mir.ndslice.slice: Slice;
+        import mir.rc.array: RCI;
+
+        MeanAccumulator!(T, Summation.precise) meanAccumulator;
         double[] circularBuffer;
         size_t frontIndex;
 
-        double avg() @property const
+        @disable this(this);
+
+        auto avg() @property const
         {
-            return summator.sum / circularBuffer.length;
+            return meanAccumulator.mean;
         }
 
         this(double[] buffer)
         {
             assert(buffer.length);
             circularBuffer = buffer;
-            summator = 0;
-            summator.put(buffer);
+            meanAccumulator.put(buffer);
         }
 
         ///operation without rounding
-        void put(double x)
+        void put(T x)
         {
             import mir.utility: swap;
-            summator += x;
+            meanAccumulator.summator += x;
             swap(circularBuffer[frontIndex++], x);
-            summator -= x;
+            meanAccumulator.summator -= x;
             frontIndex %= circularBuffer.length;
```

Comment:
btw, in case you wish to add similar API to mir-stat:
```d
// this line should be before `meanAccumulator.summator -= x;`
frontIndex = frontIndex == circularBuffer.length ? 0 : frontIndex;
meanAccumulator.summator -= x;
```
This will be a much faster them `%=`